test/test_t2star.py

```diff
@@ -0,0 +1,250 @@
+        """
+
+        configuration = QasmBackendConfiguration(
+            backend_name="t2star_simulator",
+            backend_version="0",
+            n_qubits=int(1e6),
+            basis_gates=["barrier", "h", "p", "delay", "measure"],
+            gates=[],
+            local=True,
+            simulator=True,
+            conditional=False,
+            open_pulse=False,
+            memory=False,
+            max_shots=int(1e6),
+            coupling_map=None,
+            dt=dt_factor,
+        )
+
+        self._t2star = p0["t2star"]
+        self._a_guess = p0["a_guess"]
+        self._f_guess = p0["f_guess"]
+        self._phi_guess = p0["phi_guess"]
+        self._b_guess = p0["b_guess"]
+        self._initial_prob_plus = initial_prob_plus
+        self._readout0to1 = readout0to1
+        self._readout1to0 = readout1to0
+        self._dt_factor = dt_factor
+        super().__init__(configuration)
+
+    # pylint: disable = arguments-differ
+    def run(self, qobj):
+        """
+        Run the T2star backend
+        """
+        shots = qobj.config.shots
+        result = {
+            "backend_name": "T2star backend",
+            "backend_version": "0",
+            "qobj_id": 0,
+            "job_id": 0,
+            "success": True,
+            "results": [],
+        }
+        for circ in qobj.experiments:
+            nqubits = circ.config.n_qubits
+            counts = dict()
+
+            if self._readout0to1 is None:
+                ro01 = np.zeros(nqubits)
+            else:
+                ro01 = self._readout0to1
+
+            if self._readout1to0 is None:
+                ro10 = np.zeros(nqubits)
+            else:
+                ro10 = self._readout1to0
+
+            for _ in range(shots):
+                if self._initial_prob_plus is None:
+                    prob_plus = np.ones(nqubits)
+                else:
+                    prob_plus = self._initial_prob_plus.copy()
+
+                clbits = np.zeros(circ.config.memory_slots, dtype=int)
+                for op in circ.instructions:
+                    qubit = op.qubits[0]
+
+                    if op.name == "delay":
+                        delay = op.params[0]
+                        t2star = self._t2star[qubit] * self._dt_factor
+                        freq = self._f_guess[qubit] / self._dt_factor
+
+                        prob_plus[qubit] = (
+                            self._a_guess[qubit]
+                            * np.exp(-delay / t2star)
+                            * np.cos(2 * np.pi * freq * delay + self._phi_guess[qubit])
+                            + self._b_guess[qubit]
+                        )
+
+                    if op.name == "measure":
+                        # measure in |+> basis
+                        meas_res = np.random.binomial(
+                            1,
+                            prob_plus[qubit] * (1 - ro10[qubit])
+                            + (1 - prob_plus[qubit]) * ro01[qubit],
```

Comment:
Check this expression, it is possible that there is a 0-1 confusion here

Reply:
I think you are right here.